src/main/java/org/variantsync/diffdetective/variation/tree/view/relevance/Configure.java

```diff
@@ -73,24 +76,68 @@ public Configure(final Map<String, Boolean> assignment) {
 
     @Override
     public boolean test(VariationNode<?, ?> v) {
-        return SAT.isSatisfiable(
-                FixTrueFalse.Formula.and(
-                        configuration,
-                        FixTrueFalse.EliminateTrueAndFalse(v.getPresenceCondition())
-                )
-        );
+        return ConfigureSpec.test(configuration, v);
+    }
+
+    private <TreeNode extends VariationNode<TreeNode, ?>> void computeViewNodes(List<TreeNode> vs, Consumer<TreeNode> markRelevant) {
+        for (final TreeNode c : vs) {
+            computeViewNodes(c, markRelevant);
+        }
     }
 
     @Override
     public <TreeNode extends VariationNode<TreeNode, ?>> void computeViewNodes(TreeNode v, Consumer<TreeNode> markRelevant) {
-        markRelevant.accept(v);
+        // The implementation of this method must be semantically equivalent to the default implementation Relevance.super.computeViewNodes(v, markRelevant);
+        // The implementation of this method is supposed to be optimized to avoid redundant SAT calls when possible.
+        // This requirement is modelled explicitly in terms of the ConfigureSpec class.
+
+        // If the child is an artifact, it has the same presence condition as v does, so it is also included in the view.
+        // The root must be in any view to ensure consistency.
+        if (v.isArtifact() || v.isRoot()) {
+            markRelevant.accept(v);
+            computeViewNodes(v.getChildren(), markRelevant);
+        } else {
+            computeViewNodesOfElifChain(v, markRelevant);
+        }
+    }
+
+    /**
+     * @return true if the given node was marked relevant
+     */
+    private <TreeNode extends VariationNode<TreeNode, ?>> boolean computeViewNodesOfElifChain(TreeNode v, Consumer<TreeNode> markRelevant) {
+        final NodeType vt = v.getNodeType();
+        Assert.assertTrue(vt == NodeType.IF || vt == NodeType.ELSE || vt == NodeType.ELIF);
```

Comment:
This can be simplified:
```suggestion
        Assert.assertTrue(v.isAnnotation());
```